lib/static/new-ui/experiments/time-travel/components/SnapshotsPlayer/index.tsx

```diff
@@ -166,43 +337,161 @@ export function SnapshotsPlayer(): ReactNode {
+                playerRef.current &&
+                e.timestamp <= (playerRef.current.getMetaData().startTime + newTime)
+            );
+
+            if (lastColorScheme) {
+                const colorScheme = getColorSchemeFromEvent(lastColorScheme);
+                if (colorScheme) {
+                    setIframeColorScheme(colorScheme);
+                }
+            }
+
             playerRef.current.pause(newTime);
+
+            const playerStartTime = playerRef.current.getMetaData().startTime;
+            const currentAction = findActionByTime(testSteps, playerStartTime, newTime);
+            if (currentAction && currentAction.id !== lastSetStepId.current) {
+                lastSetStepId.current = currentAction.id;
+                dispatch(setCurrentStep({stepId: currentAction.id}));
+            }
         }
-    }, []);
+    }, [testSteps]);
+    const onTimelineHover = useCallback((newTime: number) => {
+        if (playerRef.current) {
+            finishedPlayingRef.current = false;
+            const playerStartTime = playerRef.current.getMetaData().startTime;
+            const currentAction = findActionByTime(testSteps, playerStartTime, newTime);
+            if (currentAction && currentAction.id !== currentHighlightStepId) {
+                dispatch(setCurrentHighlightStep({stepId: currentAction.id}));
+            }
+        }
+    }, [testSteps]);
+    const onTimelineMouseLeave = useCallback(() => {
+        dispatch(setCurrentHighlightStep({stepId: null}));
+    }, [dispatch]);
+
+    const currentPlayerHighlightState = useSelector(state => state.app.snapshotsPlayer);
 
-    const currentHighlightPlayerTime = useSelector(state => state.app.snapshots.currentPlayerTime);
+    useEffect(() => {
+        if (currentPlayerHighlightState.isActive) {
+            playerRef.current?.pause(currentPlayerHighlightState.highlightEndTime - playerRef.current?.getMetaData().startTime);
+            setIsPlaying(false);
+            cancelTimeTicking();
+        } else if (!isPlaying) {
+            playerRef.current?.pause(currentPlayerTime > 0 ? currentPlayerTime : MIN_PLAYER_TIME);
+        }
+    }, [currentPlayerHighlightState]);
 
     useEffect(() => {
-        playerRef.current?.pause(currentHighlightPlayerTime - playerRef.current?.getMetaData().startTime);
-    }, [currentHighlightPlayerTime]);
+        if (!playerRef.current) {
+            return;
+        }
+
+        const newPlayerTime = Math.max(currentPlayerHighlightState.goToTime - playerRef.current.getMetaData().startTime, MIN_PLAYER_TIME);
+        setCurrentPlayerTime(newPlayerTime);
+        playerRef.current.pause(newPlayerTime);
+    }, [currentPlayerHighlightState.goToTime]);
+
+    const playerContainerStyle: React.CSSProperties = {
+        aspectRatio: `${maxPlayerSize.width} / ${maxPlayerSize.height}`,
+        maxWidth: `${maxPlayerSize.width}px`,
+        maxHeight: `calc(100vh - var(--sticky-header-height) - 150px)`,
```

Comment:
why is it exact 150px?

Reply:
It's just a number to add some space around the player. Added a comment.